src/niamoto/core/plugins/transformers/class_objects/categories_mapper.py

```
@@ -0,0 +1,173 @@
+                        category_values[subcategory_name] = value
+                    except (IndexError, KeyError) as e:
+                        raise DataTransformError(
+                            f"Failed to find value for {subcategory_name} in {category}",
+                            details={
+                                "class_name": class_name,
+                                "available_class_names": category_data["class_name"]
+                                .unique()
+                                .tolist(),
+                                "error": str(e),
+                            },
+                        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)